### ISSUE TITLE
ChangeWorkflow, cannot change to Chain

### ISSUE THREAD

**Issue by AugustT** [zoonproject/zoon#196]
I have come across this while writing tests. It is not possible the change a workflow using a chain. 

```r  
# Original run
  set.seed(1)
  w1 <- workflow(UKAnophelesPlumbeus,
                 UKAir, 
                 OneHundredBackground,
                 LogisticRegression,
                 SameTimePlaceMap)
  
  set.seed(1)
  w9 <- ChangeWorkflow(w1,
                       occurrence = Chain(UKAnophelesPlumbeus, UKAnophelesPlumbeus)
                       )
```

```
Error in oldCallArgs[["occurrence"]] <- as.character(occSub) : 
  more elements supplied than there are to replace
```

Looks like the function is not setup to consider chains or lists and might need quite a few changes to fix.

**Comment by timcdlucas** [comment]
Yeah, just to say `ChangeWorkflow` and `RerunWorkflow` were written quite near the end of the internship and probably haven't had the care and attention they deserve.

**Comment by AugustT** [comment]
Fixed in f6ba44b11c89618eff46b91037bf3ee81e40af14 

My approach was to align ChangeWorkflow and RerunWorkflow functions with the main workflow function as much as possible. The main issue appeared to be that on re-running/changing, the workflow element call.list was a character but the workflow functions are built to expect calls. I created a new function in zoon helpers called StringToCall to convert these into calls. This allows us to proceed exactly as in the main workflow which uses only calls.
